### PR TITLE
Improve gmtvector usage

### PR DESCRIPTION
**Pull request by PaulWessel**
Same drill as for others listed at #5341.

### CODE REVIEW

File: src/gmtvector.c [hunk 1]
```diff
@@ -112,44 +112,48 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GMTVECTOR_CTRL *C) {	/* Deal
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [<table>] [-A[m][<conf>]|<vector>] [-C[i|o]] [-E] [-N] [-S<vector>]\n", name);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-Ta|b|d|D|p<az>|r<rot>|R|s|t[az/d]|x] [%s] [%s] [%s] [%s]\n\t[%s] [%s] [%s]\n\t[%s] [%s]\n\t[%s] [%s] [%s] [%s]\n\n",
-		GMT_V_OPT, GMT_b_OPT, GMT_d_OPT, GMT_e_OPT, GMT_f_OPT, GMT_g_OPT, GMT_h_OPT, GMT_i_OPT, GMT_o_OPT, GMT_q_OPT, GMT_s_OPT, GMT_colon_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s [<table>] [-A[m][<conf>]|<vector>] [-C[i|o]] [-E] [-N] [-S<vector>] "
+		"[-Ta|b|d|D|p<azim>|r<rot>|R|s|t[<azim>/<dist>]|x] [%s] [%s] [%s] [%s] t[%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s]\n",
```
Comment (meghanrjones) [inline src/gmtvector.c:116]:
```suggestion
		"[-Ta|b|d|D|p<azim>|r<rot>|R|s|t[<azim>/<dist>]|x] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s]\n",
```

Comment (meghanrjones) [reply src/gmtvector.c:116]: -j is missing here but is in rst docs

File: doc/rst/source/gmtvector.rst [hunk 2]
```diff
@@ -12,10 +12,13 @@ Synopsis
 
 .. include:: common_SYN_OPTs.rst_
 
-**gmt vector** [ *tables* ] [ |-A|\ **m**\ [*conf*]\|\ *vector* ]
+**gmt vector** [ *tables* ]
```
Comment (meghanrjones) [inline doc/rst/source/gmtvector.rst:15]: -s common option is missing

File: src/gmtvector.c [hunk 3]
```diff
@@ -112,44 +112,48 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GMTVECTOR_CTRL *C) {	/* Deal
...
-	GMT_Message (API, GMT_TIME_NONE, "\t   -TR will instead assume the input vectors/angles are different rotations and repeatedly\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t      rotate the fixed secondary vector (see -S) using the input rotations.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   -Tt will translate input vectors to points a distance <d> away in the azimuth <az>.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t      Append <az>/<d> for a fixed set of azimuth and distance for all points,\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t      otherwise we expect to read azim, dist from the input file; append a unit [e]\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   -Tx will compute cross-product(s) with secondary vector (see -S).\n");
+	GMT_Usage (API, 1, "\n<table>");
+	GMT_Usage (API, -2, "<table> is one or more data files (in ASCII, binary, netCDF) with (x,y[,z]), (r,theta) or (lon,lat) in the "
+		"first 2-3 input columns. If one item is given and it cannot be opened we will interpret it as a single x/y[/z], r/theta, or lon/lat entry. "
+		"If no file(s) is given, standard input is read.");
```
Comment (meghanrjones) [inline src/gmtvector.c:125]: table is listed as required online

File: doc/rst/source/gmtvector.rst (hunk 2 above)
Comment (meghanrjones) [inline doc/rst/source/gmtvector.rst:15]: either here should be table or below should be tables

File: src/gmtvector.c [hunk 4]
```diff
@@ -112,44 +112,48 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GMTVECTOR_CTRL *C) {	/* Deal
...
+		"optionally append a different confidence level in percent.");
+	GMT_Usage (API, 1, "\n-C[i|o]");
+	GMT_Usage (API, -2, "Indicate Cartesian coordinates on input/output instead of lon,lat or r/theta. "
+		"Append i or o to only affect input or output coordinates.");
+	GMT_Usage (API, 1, "\n-E Automatically convert between geodetic and geocentric coordinates [no conversion]. "
+		"Ignored unless -fg is given.");
+	GMT_Usage (API, 1, "\n-N Normalize the transformed vectors (only affects -Co output).");
+	GMT_Usage (API, 1, "\n-S<vector>");
+	GMT_Usage (API, -2, "The secondary vector (if needed by -T), given as lon/lat, r/theta, or x/y[/z].");
+	GMT_Usage (API, 1, "\n-Ta|b|d|D|p<<azim>>|r<rot>|R|s|t[<azim>/<dist>]|x");
```
Comment (meghanrjones) [inline src/gmtvector.c:139]:
```suggestion
	GMT_Usage (API, 1, "\n-Ta|b|d|D|p<azim>|r<angle>|R|s|t[<azim>/<dist>]|x");
```

File: src/gmtvector.c [hunk 5]
```diff
@@ -112,44 +112,48 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GMTVECTOR_CTRL *C) {	/* Deal
...
+	GMT_Usage (API, -2, "Specify the desired transformation of the input data:");
+	GMT_Usage (API, 3, "a: Average of the primary and secondary vector (see -S).");
+	GMT_Usage (API, 3, "b: Find the bisector great circle pole(s) for input and secondary vector (see -S).");
+	GMT_Usage (API, 3, "d: Compute dot-product(s) with secondary vector (see -S).");
+	GMT_Usage (API, 3, "D: Same as -Td, but returns the angles in degrees between the vectors.");
+	GMT_Usage (API, 3, "p: Find pole to great circle with <<azim>> azimuth trend at input vector location.");
```
Comment (meghanrjones) [inline src/gmtvector.c:145]:
```suggestion
	GMT_Usage (API, 3, "p: Find pole to great circle with <azim> azimuth trend at input vector location.");
```

**Review by meghanrjones — APPROVED** [review]

File: doc/rst/source/gmtvector.rst (hunk 2 above)
Comment (PaulWessel) [reply doc/rst/source/gmtvector.rst:15]: Added

File: src/gmtvector.c (hunk 3 above)
Comment (PaulWessel) [reply src/gmtvector.c:125]: OK, now also in the usage.

**Pull request merged**